packages/tools/src/tools/displayTools/Labelmap/labelmapDisplay.ts

```diff
@@ -134,6 +135,61 @@ function removeSegmentationRepresentation(
   }
 }
 
+/**
+ * Checks if a segmentation data belongs to the same study as the series
+ * displayed in a given viewport
+ * @param viewport
+ * @param referencedVolumeId volume id of the segmentation reference series
+ * @returns
+ */
+function canRenderSegmentationInViewport(
+  viewport,
+  referencedVolumeId,
+  restrictVisualization
+) {
+  const defaultActor = viewport.getDefaultActor();
+  if (defaultActor) {
+    const { uid: defaultActorUID } = defaultActor;
+    const volume = cache.getVolume(defaultActorUID);
+    const viewportImageIds = volume.imageIds;
+
+    if (viewportImageIds) {
+      const referencedVolume = cache.getVolume(referencedVolumeId);
+      const segmentationImageIds = referencedVolume.imageIds;
+      if (segmentationImageIds) {
+        const segmentationSeriesInfo = metaData.get(
+          'generalSeriesModule',
+          segmentationImageIds[0]
+        );
+        const segmentationImageInfo = metaData.get(
+          'imagePlaneModule',
+          segmentationImageIds[0]
+        );
+        const viewportSeriesInfo = metaData.get(
+          'generalSeriesModule',
+          viewportImageIds[0]
+        );
+        const viewportImageInfo = metaData.get(
+          'imagePlaneModule',
+          viewportImageIds[0]
+        );
+        if (
+          segmentationSeriesInfo.studyInstanceUID ===
```

Comment:
This restricts the viewing to same study instance UID, NOT the same series instance UID as is specified elsewhere.